.github/workflows/java-ci.yml

```diff
@@ -0,0 +1,150 @@
+
+      # 3. Cache Maven dependencies
+      - name: Cache Maven Dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-
+
+      # 4. Build the project with Gradle
+      - name: Build with Gradle
+        run: |
+          cd diff-java-tool
+          ./gradlew clean build
+
+      # 5. Copy DiffTool JAR to workspace
+      - name: Copy DiffTool JAR to workspace
+        run: |
+          cp diff-java-tool/build/libs/diff-java-tool-*-all.jar "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+
+      # 6. Download patch-diff-report-tool JAR
+      - name: Download patch-diff-report-tool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "checkstyle/contribution"
+          tag: "patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}"
+          fileName: >-
+            patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar
+          out-file-path: "${{ github.workspace }}"
+
+      # 7. Clone Checkstyle Repository
+      - name: Clone Checkstyle Repository
+        run: |
+          git clone https://github.com/checkstyle/checkstyle.git "${{ github.workspace }}/checkstyle"
+
+      # 8. Create Temporary Patch Branch
+      - name: Create Temporary Patch Branch
+        run: |
+          cd "${{ github.workspace }}/checkstyle"
+          git config user.email "github-actions@github.com"
+          git config user.name "GitHub Actions"
+          git checkout -b temp-patch-branch
+          # Optionally make a minor change to ensure reports are different
+          echo "// Temporary change" >> dummy.java
+          git add dummy.java
+          git commit -m "Temporary change for testing"
+
+      # 9. Execute DiffTool.jar in 'diff' mode
+      - name: Execute DiffTool.jar in Diff Mode
+        env:
+          CONFIG_XML: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/config.xml"
+          PROJECTS_PROPERTIES: "${{ github.workspace }}/${{ env.CONFIG_DIR }}/project.properties"
+          DIFFTOOL_JAR_PATH: "${{ github.workspace }}/${{ env.DIFFTOOL_JAR }}"
+          PATCH_DIFF_TOOL_JAR_PATH: "${{ github.workspace }}/patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar"
+        run: |
+          REPO="${{ github.workspace }}/checkstyle"
+
+          echo "Running DiffTool.jar in 'diff' mode"
+          java -jar "${DIFFTOOL_JAR_PATH}" \
+            --localGitRepo "$REPO" \
+            --baseBranch master \
+            --config "${CONFIG_XML}" \
+            --patchBranch temp-patch-branch \
+            --listOfProjects "${PROJECTS_PROPERTIES}" \
+            --allowExcludes \
+            --diffToolJarPath "${PATCH_DIFF_TOOL_JAR_PATH}" || { echo "DiffTool execution failed"; exit 1; }
+
+      # 10. Execute DiffTool.jar with all arguments
+      - name: Execute DiffTool.jar with all arguments
```

Comment:
With patch config.

Reply:
done

Reply:
Not done, name is still the same

Reply:
done now.

Reply:
I still see name `Execute DiffTool.jar with all arguments`
should be  `Execute DiffTool.jar with patch config`

Reply:
do the same for other run.

Reply:
> do the same for other run.

 - name: Execute DiffTool.jar in Diff Mode rename it to Execute DiffTool.jar with config
 Is that it?

Reply:
> I still see name `Execute DiffTool.jar with all arguments` should be `Execute DiffTool.jar with patch config`

done